src/js/visual/image.js

```diff
@@ -269,6 +269,7 @@ ripe.Image.prototype._registerHandlers = function() {
 ripe.Image.prototype._unregisterHandlers = function() {
     if (this.loadListener) this.element.removeEventListener("load", this.loadListener);
     if (this.errorListener) this.element.removeEventListener("error", this.errorListener);
+    if (this._loadedCallback) this._loadedCallback({ canceled: true });

```

Comment:
@NFSS10 this should probably be `this.cancel()`, no? Right now it's probably the same thing, but in the future it might contain more stuff.

Besides, I'd place that call under `deinit` instead. Right now it's under `unregisterHandlers` which is about unregistering event listeners, so it seems out of place.

After this please review `configurator` for the same class of problems, if I had to guess the same should be happening there, it's just not a problem because people are never calling `deinit` on them.

Reply:
The hard work is done, we're in the last mile 💪